parsl/executors/radical/executor.py

```diff
@@ -0,0 +1,583 @@
+        logger.info('Parsl: {0}'.format(parsl.__version__))
+        logger.info('RADICAL pilot: {0}'.format(rp.version))
+        self.session = rp.Session(cfg={'base': self.run_dir},
+                                  uid=ru.generate_id('rpex.session',
+                                                     mode=ru.ID_PRIVATE))
+        logger.info("RPEX session is created: {0}".format(self.session.path))
+
+        pd_init = {'gpus': self.gpus,
+                   'cores': self.cores,
+                   'exit_on_error': True,
+                   'queue': self.partition,
+                   'project': self.project,
+                   'runtime': self.walltime,
+                   'resource': self.resource,
+                   'access_schema': self.access_schema}
+
+        if not self.resource or 'local' in self.resource:
+            # move the agent sandbox to the workdir mainly
+            # for debugging purposes. This will allow parsl
+            # to include the agent sandbox with the ci artifacts.
+            if os.environ.get("LOCAL_SANDBOX"):
+                pd_init['sandbox'] = self.run_dir
+                os.environ["RADICAL_LOG_LVL"] = "DEBUG"
+
+            logger.info("RPEX will be running in the local mode")
+
+        self.rpex_cfg = self.rpex_cfg._get_cfg_file(path=self.run_dir)
+        cfg = ru.Config(cfg=ru.read_json(self.rpex_cfg))
+
+        self.master = cfg.master_descr
+        self.n_masters = cfg.n_masters
+
+        pd = rp.PilotDescription(pd_init)
+
+        tds = list()
+        master_path = '{0}/rpex_master.py'.format(PWD)
+        worker_path = '{0}/rpex_worker.py'.format(PWD)
+
+        for i in range(self.n_masters):
+            td = rp.TaskDescription(self.master)
+            td.mode = rp.RAPTOR_MASTER
+            td.uid = ru.generate_id('master.%(item_counter)06d', ru.ID_CUSTOM,
+                                    ns=self.session.uid)
+            td.ranks = 1
+            td.cores_per_rank = 1
+            td.arguments = [self.rpex_cfg, i]
+            td.input_staging = self._stage_files([File(master_path),
+                                                  File(worker_path),
+                                                  File(self.rpex_cfg)], mode='in')
+            tds.append(td)
+
+        self.pmgr = rp.PilotManager(session=self.session)
+        self.tmgr = rp.TaskManager(session=self.session)
+
+        # submit pilot(s)
+        pilot = self.pmgr.submit_pilots(pd)
+        self.tmgr.submit_tasks(tds)
+
+        # prepare or use the current env for the agent/pilot side environment
+        if cfg.pilot_env_mode != 'client':
+            logger.info("creating {0} environment for the executor".format(cfg.pilot_env.name))
+            pilot.prepare_env(env_name=cfg.pilot_env.name,
+                              env_spec=cfg.pilot_env.as_dict())
+        else:
+            client_env = sys.prefix.split
+            logger.info("reusing ({0}) environment for the executor".format(client_env))
```

Comment:
This looks weird on my laptop:

```
INFO     parsl.executors.radical.executor:executor.py:285 reusing (<built-in method split of str object at 0x7fb63c646c10>) environment for the executor
```

Reply:
sorry I fixed it: this should be like this now:
```
 parsl.executors.radical.executor:executor.py:285 reusing (/home/aymen/.venv) environment for the executor
```